### DISCUSSION TITLE
[Adonis 4.1]Check if Relationship exists between two model

### DISCUSSION THREAD

**Discussion by tayfunakgc** [adonisjs/core#1484] [Help]
I have two models : User and Organization. There is many to many relationship between them.
Before showing an organization by id, I want to check the current user has the organization by id in a middleware.

In show method; I use 'findOrFail' to get organization (/orgs/:id, OrgController.show)  
Before that I must to check the user has this organization. How can I achieve that?
Is there any method that returns bool if the current user has the organization?

```
//* Org Model
users () {
    return this.belongsToMany('App/Models/User')
        .pivotTable('org_users');
}
```
```
//* User Model
orgs() {
    return this.belongsToMany('App/Models/Org')
        .pivotTable('org_users');
}
```

**Comment by popovicieduard** [comment]
I think you can make use of https://adonisjs.com/docs/4.1/relationships#_wherehas
```
const org= await Org
  .query()
  .whereHas('users', (builder) => {
    builder.where('id', userId)
  })
  .findByOrFail('id', orgId)
```
